components/tikv_util/src/codec/bytes.rs

```diff
@@ -278,6 +278,57 @@ pub fn decode_bytes_in_place(data: &mut Vec<u8>, desc: bool) -> Result<()> {
     }
 }
 
+/// Returns whether `encoded` bytes is encoded from `raw`.
+///
+/// # Panics
+///
+/// Panics if `encoded` is not valid
+pub fn is_encoded_from(encoded: &[u8], raw: &[u8], desc: bool) -> bool {
+    let check_single_chunk = |encoded: &[u8], raw: &[u8]| {
+        let len = raw.len();
+        let pad = (ENC_GROUP_SIZE - len) as u8;
+        if desc {
+            encoded[ENC_GROUP_SIZE] == !(ENC_MARKER - pad)
+                && encoded[..len]
+                    .iter()
+                    .zip(raw)
+                    .all(|(&enc, &raw)| enc == !raw)
+                && encoded[len..encoded.len() - 1].iter().all(|&v| v == 0xff)
+        } else {
+            encoded[ENC_GROUP_SIZE] == (ENC_MARKER - pad)
+                && &encoded[..len] == raw
+                && encoded[len..encoded.len() - 1].iter().all(|&v| v == 0)
+        }
+    };
+
+    let mut rev_encoded_chunks = encoded.rchunks_exact(ENC_GROUP_SIZE + 1);
+    // Valid encoded bytes must has complete chunks
+    assert!(rev_encoded_chunks.remainder().is_empty());
+
+    // Bytes are compared in reverse order because in real cases like TiDB, if two keys
+    // are different, the last a few bytes are more likely to be different.
+
+    let raw_chunks = raw.chunks_exact(ENC_GROUP_SIZE);
+    // Check the last chunk first
+    match rev_encoded_chunks.next() {
+        Some(encoded_chunk) if check_single_chunk(encoded_chunk, raw_chunks.remainder()) => {}
```

Comment:
what will happen if `raw_chunks`'s length is `N*ENC_GROUP_SIZE` so that `raw_chunks.remainder()` is empty?

Reply:
Then the encoded chunk must be `00 00 00 00 00 00 00 00 FF`